remappers/hostmetrics/filesystem.go

```diff
@@ -0,0 +1,149 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package hostmetrics
+
+import (
+	"strings"
+
+	remappers "github.com/elastic/opentelemetry-lib/remappers/internal"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"golang.org/x/exp/constraints"
+)
+
+func remapFilesystemMetrics(src, out pmetric.MetricSlice,
+	_ pcommon.Resource,
+	dataset string,
+) error {
+	var timestamp pcommon.Timestamp
+	var device, mpoint, fstype string
+	totalUsagePerDevice := make(map[string]int64)
+	totalInodeUsagePerDevice := make(map[string]int64)
+	usedBytesPerDevice := make(map[string]int64)
+
+	for i := 0; i < src.Len(); i++ {
+		metric := src.At(i)
+		switch metric.Name() {
+		case "system.filesystem.usage", "system.filesystem.inodes.usage":
+			dataPoints := metric.Sum().DataPoints()
+			for j := 0; j < dataPoints.Len(); j++ {
+				dp := dataPoints.At(j)
+				value := dp.IntValue()
+				timestamp = dp.Timestamp()
+				deviceValue, mpointValue, fstypeValue, ok := getAttributes(dp)
+				if !ok {
+					continue
+				}
+				device, mpoint, fstype = deviceValue.Str(), mpointValue.Str(), fstypeValue.Str()
```

Comment:
```suggestion
				device, mpoint, fstype, ok := getAttributes(dp)
				if !ok {
					continue
				}
```
I think we can directly use the returned values for `device`, `mpoint`, and `fstype` instead of explicitly converting them to string type as in L52. This was required because these were declared in L34 as strings. That shouldn't be required if they are not returned or required to be used outside the `remapFilesystemMetrics` function.

Reply:
The device, mpoint, fstype are being used as strings in multiple places (be it within the function) while adding the metrics. Hence I would prefer keeping it this way.

Reply:
The are used as strings that's why it's better to keep them close to wherever they are being used instead of declaring them at the start of the function. Leave the choice to you.

Reply:
+1, there is no point in keeping these vars at a higher scope than the for loop.